group-coordinator/src/test/java/org/apache/kafka/coordinator/group/GroupMetadataManagerTest.java

```diff
@@ -2377,156 +2606,2165 @@ public void testOnNewMetadataImage() {
+            .withSessionTimeoutMs(5000)
+            .withRebalanceTimeoutMs(5000)
+            .build();
+
+        JoinGroupResponseData response = context.joinGenericGroup(request, false, true);
+        assertEquals(Errors.NONE.code(), response.errorCode());
+
+        CompletableFuture<JoinGroupResponseData> responseFuture = context.sendGenericGroupJoin(request.setMemberId("unknown-member-id"));
+        assertTrue(responseFuture.isDone());
+        assertEquals(Errors.FENCED_INSTANCE_ID.code(), responseFuture.get(5, TimeUnit.SECONDS).errorCode());
+    }
+
+    @Test
+    public void testJoinGroupUnknownConsumerExistingGroup() {
+        GroupMetadataManagerTestContext context = new GroupMetadataManagerTestContext.Builder()
+            .build();
+
+        JoinGroupRequestData request = new JoinGroupRequestBuilder()
+            .withGroupId("group-id")
+            .withMemberId(UNKNOWN_MEMBER_ID)
+            .withDefaultProtocolTypeAndProtocols()
+            .withSessionTimeoutMs(5000)
+            .withRebalanceTimeoutMs(5000)
+            .build();
+
+        JoinGroupResponseData response = context.joinGenericGroupAsDynamicMember(request);
+        assertEquals(Errors.NONE.code(), response.errorCode());
+
+        JoinGroupResponseData otherResponse = context.joinGenericGroupAsDynamicMember(request
+            .setMemberId("other-member-id"));
+        assertEquals(Errors.UNKNOWN_MEMBER_ID.code(), otherResponse.errorCode());
+    }
+
+    @Test
+    public void testJoinGroupUnknownConsumerNewDeadGroup() throws Exception {
+        GroupMetadataManagerTestContext context = new GroupMetadataManagerTestContext.Builder()
+            .build();
+
+        GenericGroup group = context.groupMetadataManager.getOrMaybeCreateGenericGroup(
+            "group-id",
+            true
+        );
+        group.transitionTo(DEAD);
+
+        JoinGroupRequestData request = new JoinGroupRequestBuilder()
+            .withGroupId("group-id")
+            .withMemberId(UNKNOWN_MEMBER_ID)
+            .withDefaultProtocolTypeAndProtocols()
+            .build();
+
+        CompletableFuture<JoinGroupResponseData> responseFuture = context.sendGenericGroupJoin(request);
+        assertTrue(responseFuture.isDone());
+        assertEquals(Errors.COORDINATOR_NOT_AVAILABLE.code(), responseFuture.get(5, TimeUnit.SECONDS).errorCode());
+    }
+
+    @Test
+    public void testJoinGroupProtocolTypeIsNotProvidedWhenAnErrorOccurs() {
+        GroupMetadataManagerTestContext context = new GroupMetadataManagerTestContext.Builder()
+            .build();
+
+        JoinGroupRequestData request = new JoinGroupRequestBuilder()
+            .withGroupId("group-id")
+            .withMemberId("member-id")
+            .withDefaultProtocolTypeAndProtocols()
+            .build();
+
+        JoinGroupResponseData response = context.joinGenericGroupAsDynamicMember(request);
+        assertEquals(Errors.GROUP_ID_NOT_FOUND.code(), response.errorCode());
```

Comment:
There is `UNKNOWN_MEMBER_ID` in scala?

Reply:
related to when a group is not found. have reverted and updated the error code